### ISSUE TITLE
tf.keras.layers.MaxPooling3D crashes

### ISSUE THREAD

**Issue by lugalUrim** [tensorflow/tensorflow#51936]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.6.0
- Python version: 3.6.8
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

**Describe the current behavior**
`tf.keras.layers.MaxPooling3D` crashes when `pool_size` contains `0`, and outputs a all-inf tensor when `pool_size` contains negative values.

**Describe the expected behavior**
Expect a `ValueError` to be thrown if the input `pool_size` contains zero or negative values.

**Standalone code to reproduce the issue**
If the `pool_size` has `0`:
```
import tensorflow as tf
pool_size = [2, 2, 0]
layer = tf.keras.layers.MaxPooling3D(strides=1, pool_size=pool_size)
input_tensor = tf.random.uniform([3, 4, 10, 11, 12], dtype=tf.float32)
res = layer(input_tensor) # crash
```
Outputs:
```
Floating point exception (core dumped)
```
If the `pool_size` has negative values:
```
import tensorflow as tf
pool_size = [2, 2, -2]
layer = tf.keras.layers.MaxPooling3D(strides=1, pool_size=pool_size,)
input_tensor = tf.random.uniform([3, 4, 10, 11, 12], dtype=tf.float32)
res = layer(input_tensor)
print(res)
```
The output is a tensor with `shape`=`(3, 3, 9, 14, 12)` and all `inf` values.

**Comment by lugalUrim** [comment]
It turns out that `tf.keras.layers.AveragePooling3D` also crashes when `pool_size` contains `0`.

**Comment by sushreebarsa** [comment]
@lugalUrim Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!

**Comment by WingsBrokenAngel** [comment]
Hi, I have created a [PR](https://github.com/keras-team/keras/pull/15356) for Keras to solve this issue.

**Comment by yongtang** [comment]
While the original issue is coming from tf.keras.layers.MaxPooling3D, the issue is triggered when max_pool3d is called directly with tensorflow itself. For that PR #51975 has been created to fix the issue inside tensorflow.

**Comment by lugalUrim** [comment]
Thanks for the information @sushreebarsa, I will post new issues in the keras repo, but I guess we can keep this issue in tensorflow, as mentioned by @yongtang there are related bugs in the tensorflow operator implementations as well.

**Comment by lugalUrim** [comment]
Thank you @WingsBrokenAngel and @yongtang for your PRs! Actually here are more crash bugs I find when `pool_size`/`k_size` <= 0, so we may want to fix all of these in the codebase and also cover them in the unit tests.
- `tf.keras.layers.AveragePooling1D`, `tf.keras.layers.AveragePooling2D`, `tf.keras.layers.AveragePooling3D`
- `tf.keras.layers.MaxPooling1D`, `tf.keras.layers.MaxPooling2D`, `tf.keras.layers.MaxPooling3D`
- `tf.nn.avg_pool1d`, `tf.nn.avg_pool2d`, `tf.nn.avg_pool3d`
- `tf.nn.max_pool1d`, `tf.nn.max_pool2d`, `tf.nn.max_pool3d`

**Comment by oawxkw** [comment]
I also observed the following API aliases or similar APIs can cause the same issue in some older versions of tensorflow.
Users should be cautious when using them on CPU up to tensorflow 2.6.0 (v2.6.0-rc2-32-g919f693420e).

> Besides, in tensorflow 2.6.0 (v2.6.0-rc2-32-g919f693420e) and previous versions, it throws the InternalError Exception on GPU, which is acceptable.

- `tf.keras.layers.MaxPooling3D`, `tf.keras.layers.MaxPool3D`, `tf.compat.v1.keras.layers.MaxPooling3D`, `tf.compat.v1.keras.layers.MaxPool3D`
- `tf.keras.layers.AveragePooling3D`, `tf.keras.layers.AvgPool3D`, `tf.compat.v1.keras.layers.AveragePooling3D`, `tf.compat.v1.keras.layers.AvgPool3D`

It seems to be fixed in tensorflow 2.6.1 (v2.6.0-101-g3aa40c3ce9d) and later versions.
Note that, in tensorflow 2.6.1 (v2.6.0-101-g3aa40c3ce9d), it throws the InvalidArgumentError Exception, and in later versions, it throws the ValueError Exception, which is expected.
